In [11]:
suppressPackageStartupMessages({
    library(ArchR) 
    library(data.table)
    library(purrr)
    library(parallel)
    library(dplyr)
  #  library(ggpubr)
    library(Matrix)
    library(SingleCellExperiment)
    library(Seurat)
    library(reshape2)
    library(scater)
    library(viridis)
    library(reticulate)
})

options(repr.plot.width=15, repr.plot.height=8)

In [2]:
# I/O
io = list()
io$basedir='/rds/project/rds-SDzz0CATGms/users/bt392/04_Rabbit_ATAC_2'
io$output.directory <- file.path(io$basedir,"ArchR")
io$plot.dir = file.path(io$output.directory,'Plots')
io$outdir =  file.path(io$output.directory,"Motifs")

io$archR.directory = file.path(io$output.directory, 'Project/')
dir.create(file.path(io$archR.directory, 'PeakCalls'), showWarnings = FALSE)
dir.create(file.path(io$archR.directory, 'Annotations'), showWarnings = FALSE)
dir.create(file.path(io$archR.directory, 'Plots'), showWarnings = FALSE)

setwd(io$output.directory)

In [3]:
io$archR.directory = file.path(io$output.directory, 'Project/')
ArchRProject.filt = loadArchRProject(io$archR.directory)

Successfully loaded ArchRProject!


                                                   / |
                                                 /    \
            .                                  /      |.
            \\\                              /        |.
              \\\                          /           `|.
                \\\                      /              |.
                  \                    /                |\
                  \\#####\           /                  ||
                ==###########>      /                   ||
                 \\##==......\    /                     ||
            ______ =       =|__ /__                     ||      \\\
        ,--' ,----`-,__ ___/'  --,-`-===================##========>
       \               '        ##_______ _____ ,--,__,=##,__   ///
        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/
        -,____,---'       \\####\\________________,--\\_##,/
           ___      .______        ______  __    __  .____

In [5]:
ArchRProject.filt <- addGroupCoverages(ArchRProj = ArchRProject.filt, groupBy = "celltype_manual")

ArchR logging to : ArchRLogs/ArchR-addGroupCoverages-198b63663d034-Date-2021-12-12_Time-11-43-12.log
If there is an issue, please report to github with logFile!

CYT (1 of 13) : CellGroups N = 2

Early_Amniotic_Ectoderm (2 of 13) : CellGroups N = 2

Embryo_proper (3 of 13) : CellGroups N = 5

Endothelium (4 of 13) : CellGroups N = 5

Epiblast (5 of 13) : CellGroups N = 3

Erythrocytes (6 of 13) : CellGroups N = 5

Hypoblast (7 of 13) : CellGroups N = 2

Megakaryocytes (8 of 13) : CellGroups N = 2

Mesoderm (9 of 13) : CellGroups N = 2

Primitive_Streak (10 of 13) : CellGroups N = 2

SCT (11 of 13) : CellGroups N = 2

Trophoblast (12 of 13) : CellGroups N = 2

Visceral_YS_Endoderm (13 of 13) : CellGroups N = 3

2021-12-12 11:43:43 : Creating Coverage Files!, 0.52 mins elapsed.

2021-12-12 11:43:43 : Batch Execution w/ safelapply!, 0.52 mins elapsed.

2021-12-12 11:43:43 : Group CYT._.rabbit_BGRGP3 (1 of 37) : Creating Group Coverage File : CYT._.rabbit_BGRGP3.insertions.coverage.h5, 0.5

Coverage File Exists!

Added Coverage Group

Added Metadata Group

Added ArrowCoverage Class

Added Coverage/Info

Added Coverage/Info/CellNames

2021-12-12 11:58:12 : Group Hypoblast._.Rep1 (23 of 37) : Creating Group Coverage File : Hypoblast._.Rep1.insertions.coverage.h5, 15.007 mins elapsed.

Number of Cells = 500

Coverage File Exists!

Added Coverage Group

Added Metadata Group

Added ArrowCoverage Class

Added Coverage/Info

Added Coverage/Info/CellNames

2021-12-12 11:58:55 : Group Hypoblast._.Rep2 (24 of 37) : Creating Group Coverage File : Hypoblast._.Rep2.insertions.coverage.h5, 15.723 mins elapsed.

Number of Cells = 40

Coverage File Exists!

Added Coverage Group

Added Metadata Group

Added ArrowCoverage Class

Added Coverage/Info

Added Coverage/Info/CellNames

2021-12-12 12:00:53 : Group Megakaryocytes._.Rep1 (25 of 37) : Creating Group Coverage File : Megakaryocytes._.Rep1.insertions.coverage.h5, 17.685 mins elapsed.

Number of Cells = 355

Coverage File Exists!

Added

Coverage File chr11 (18 of 37)

Coverage File chr11 (19 of 37)

Coverage File chr11 (20 of 37)

Coverage File chr11 (21 of 37)

Coverage File chr11 (22 of 37)

Coverage File chr11 (23 of 37)

Coverage File chr11 (24 of 37)

Coverage File chr11 (25 of 37)

Coverage File chr11 (26 of 37)

Coverage File chr11 (27 of 37)

Coverage File chr11 (28 of 37)

Coverage File chr11 (29 of 37)

Coverage File chr11 (30 of 37)

Coverage File chr11 (31 of 37)

Coverage File chr11 (32 of 37)

Coverage File chr11 (33 of 37)

Coverage File chr11 (34 of 37)

Coverage File chr11 (35 of 37)

Coverage File chr11 (36 of 37)

Coverage File chr11 (37 of 37)

Kmer Bias chr12 (4 of 22)

chr12 
Coverage File chr12 (1 of 37)

Coverage File chr12 (2 of 37)

Coverage File chr12 (3 of 37)

Coverage File chr12 (4 of 37)

Coverage File chr12 (5 of 37)

Coverage File chr12 (6 of 37)

Coverage File chr12 (7 of 37)

Coverage File chr12 (8 of 37)

Coverage File chr12 (9 of 37)

Coverage File chr12 (10 of 37)

Coverage File c

Coverage File chr18 (10 of 37)

Coverage File chr18 (11 of 37)

Coverage File chr18 (12 of 37)

Coverage File chr18 (13 of 37)

Coverage File chr18 (14 of 37)

Coverage File chr18 (15 of 37)

Coverage File chr18 (16 of 37)

Coverage File chr18 (17 of 37)

Coverage File chr18 (18 of 37)

Coverage File chr18 (19 of 37)

Coverage File chr18 (20 of 37)

Coverage File chr18 (21 of 37)

Coverage File chr18 (22 of 37)

Coverage File chr18 (23 of 37)

Coverage File chr18 (24 of 37)

Coverage File chr18 (25 of 37)

Coverage File chr18 (26 of 37)

Coverage File chr18 (27 of 37)

Coverage File chr18 (28 of 37)

Coverage File chr18 (29 of 37)

Coverage File chr18 (30 of 37)

Coverage File chr18 (31 of 37)

Coverage File chr18 (32 of 37)

Coverage File chr18 (33 of 37)

Coverage File chr18 (34 of 37)

Coverage File chr18 (35 of 37)

Coverage File chr18 (36 of 37)

Coverage File chr18 (37 of 37)

Kmer Bias chr19 (11 of 22)

chr19 
Coverage File chr19 (1 of 37)

Coverage File chr19 (2 of 37)

Coverag

Coverage File chr5 (6 of 37)

Coverage File chr5 (7 of 37)

Coverage File chr5 (8 of 37)

Coverage File chr5 (9 of 37)

Coverage File chr5 (10 of 37)

Coverage File chr5 (11 of 37)

Coverage File chr5 (12 of 37)

Coverage File chr5 (13 of 37)

Coverage File chr5 (14 of 37)

Coverage File chr5 (15 of 37)

Coverage File chr5 (16 of 37)

Coverage File chr5 (17 of 37)

Coverage File chr5 (18 of 37)

Coverage File chr5 (19 of 37)

Coverage File chr5 (20 of 37)

Coverage File chr5 (21 of 37)

Coverage File chr5 (22 of 37)

Coverage File chr5 (23 of 37)

Coverage File chr5 (24 of 37)

Coverage File chr5 (25 of 37)

Coverage File chr5 (26 of 37)

Coverage File chr5 (27 of 37)

Coverage File chr5 (28 of 37)

Coverage File chr5 (29 of 37)

Coverage File chr5 (30 of 37)

Coverage File chr5 (31 of 37)

Coverage File chr5 (32 of 37)

Coverage File chr5 (33 of 37)

Coverage File chr5 (34 of 37)

Coverage File chr5 (35 of 37)

Coverage File chr5 (36 of 37)

Coverage File chr5 (37 of 37)

Kmer Bias ch

In [6]:
saveArchRProject(ArchRProj = ArchRProject.filt)

Saving ArchRProject...

Loading ArchRProject...

Successfully loaded ArchRProject!


                                                   / |
                                                 /    \
            .                                  /      |.
            \\\                              /        |.
              \\\                          /           `|.
                \\\                      /              |.
                  \                    /                |\
                  \\#####\           /                  ||
                ==###########>      /                   ||
                 \\##==......\    /                     ||
            ______ =       =|__ /__                     ||      \\\
        ,--' ,----`-,__ ___/'  --,-`-===================##========>
       \               '        ##_______ _____ ,--,__,=##,__   ///
        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/
        -,____,---'       \\####\\________________,--\\_##,/
         

class: ArchRProject 
outputDirectory: /rds/project/rds-SDzz0CATGms/users/bt392/04_Rabbit_ATAC_2/ArchR/Project 
samples(8): rabbit_BGRGP1 rabbit_BGRGP2 ... rabbit_BGRGP7 rabbit_BGRGP8
sampleColData names(1): ArrowFiles
cellColData names(23): Clusters sample ... stage_clusters
  celltype_manual
numberOfCells(1): 33115
medianTSS(1): 3.407
medianFrags(1): 31914

In [12]:
pathToMacs2 <- findMacs2()

# Find peaks
test <- addReproduciblePeakSet(
    ArchRProj = ArchRProject.filt, 
    groupBy = "celltype_manual", 
    pathToMacs2 = pathToMacs2,
    peakMethod = "Macs2")


Searching For MACS2..

Found with $path!

ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-198b6738b2cd-Date-2021-12-12_Time-12-34-33.log
If there is an issue, please report to github with logFile!

Calling Peaks with Macs2

2021-12-12 12:34:33 : Peak Calling Parameters!, 0.005 mins elapsed.



                                          Group nCells nCellsUsed nReplicates
CYT                                         CYT   1228       1000           2
Early_Amniotic_Ectoderm Early_Amniotic_Ectoderm    237        237           2
Embryo_proper                     Embryo_proper  14482       2118           5
Endothelium                         Endothelium    849        739           5
Epiblast                               Epiblast   6716       1500           3
Erythrocytes                       Erythrocytes   1124        719           5
Hypoblast                             Hypoblast   3068        540           2
Megakaryocytes                   Megakaryocytes    395        395           2
Mesoderm                               Mesoderm   1593       1000           2
Primitive_Streak               Primitive_Streak   1524        540           2
SCT                                         SCT    446        445           2
Trophoblast                         Trophoblast    577        54

2021-12-12 12:34:33 : Batching Peak Calls!, 0.006 mins elapsed.

2021-12-12 12:34:33 : Batch Execution w/ safelapply!, 0 mins elapsed.

2021-12-12 12:34:33 : Group 1 of 37, Calling Peaks with MACS2!, 0 mins elapsed.



ERROR: Error in (function (bedFile = NULL, pathToMacs2 = "macs2", genomeSize = 2.7e+09, : !is.null(genomeSize) is not TRUE


In [10]:
pathToMacs2

[1] "macs2"

In [14]:
ArchRProject.filt <- addReproduciblePeakSet(
    ArchRProj = ArchRProject.filt, 
    groupBy = "celltype_manual",
    peakMethod = "Tiles",
    method = "p"
)

ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-198b64123e313-Date-2021-12-12_Time-12-35-50.log
If there is an issue, please report to github with logFile!

Calling Peaks with TileMatrix. We recommend using the Macs2 Version.
This method is still under development.

Group Coverages Already Computed Returning Groups, Set force = TRUE to Recompute!



List of length 13
names(13): CYT Early_Amniotic_Ectoderm ... Trophoblast Visceral_YS_Endoderm


2021-12-12 12:35:50 : Peak Calling Parameters!, 0.005 mins elapsed.



                                          Group nCells nCellsUsed nReplicates
CYT                                         CYT   1228       1000           2
Early_Amniotic_Ectoderm Early_Amniotic_Ectoderm    237        237           2
Embryo_proper                     Embryo_proper  14482       2118           5
Endothelium                         Endothelium    849        739           5
Epiblast                               Epiblast   6716       1500           3
Erythrocytes                       Erythrocytes   1124        719           5
Hypoblast                             Hypoblast   3068        540           2
Megakaryocytes                   Megakaryocytes    395        395           2
Mesoderm                               Mesoderm   1593       1000           2
Primitive_Streak               Primitive_Streak   1524        540           2
SCT                                         SCT    446        445           2
Trophoblast                         Trophoblast    577        54

2021-12-12 12:36:01 : Computing Total Accessibility Across All Features, 0.187 mins elapsed.

2021-12-12 12:36:06 : Computing Pseudo-Grouped Tile Matrix, 0.274 mins elapsed.

2021-12-12 12:40:50 : Created Pseudo-Grouped Tile Matrix (1.583 GB), 5.004 mins elapsed.

colSums = 22150790colSums = 23345998colSums = 8292839colSums = 1916852colSums = 25425278colSums = 22269836colSums = 23838001colSums = 24553874colSums = 4025482colSums = 5891137colSums = 5508313colSums = 6763335colSums = 5731092colSums = 5470166colSums = 17787111colSums = 24571551colSums = 25334484colSums = 11327926colSums = 2213859colSums = 1866361colSums = 1928997colSums = 2203756colSums = 14087394colSums = 1943859colSums = 7905571colSums = 1440218colSums = 22242573colSums = 23619986colSums = 13878963colSums = 1165601colSums = 13360248colSums = 12365816colSums = 18581099colSums = 1700454colSums = 13132912colSums = 13283693colSums = 7823453

nTiles = 4495514

Expectation = 4.92730975812777Expectation = 5.19317657558179Expecta

Warning message in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE):
“GRanges object contains 3 out-of-bound ranges located on sequences
  chr12, chr13, and chr16. Note that ranges located on a sequence whose
  length is unknown (NA) or on a circular sequence are not considered
  out-of-bound (use seqlengths() and isCircular() to get the lengths and
  circularity flags of the underlying sequences). You can use trim() to
  trim these ranges. See ?`trim,GenomicRanges-method` for more
  information.”
Annotating Peaks : TSS

Annotating Peaks : GC

Annotating Peaks : Nearest Gene

Annotating Peaks : Gene

Warning message in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE):
“GRanges object contains 3 out-of-bound ranges located on sequences
  chr12, chr13, and chr16. Note that ranges located on a sequence whose
  length is unknown (NA) or on a circular sequence are not considered
  out-of-bound (use seqlengths() and isCircular() to get the lengths and
  circularity flags of the underly

Annotating Peaks : TSS

Annotating Peaks : GC

Annotating Peaks : Nearest Gene

Annotating Peaks : Gene

Warning message in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE):
“GRanges object contains 3 out-of-bound ranges located on sequences
  chr12, chr13, and chr16. Note that ranges located on a sequence whose
  length is unknown (NA) or on a circular sequence are not considered
  out-of-bound (use seqlengths() and isCircular() to get the lengths and
  circularity flags of the underlying sequences). You can use trim() to
  trim these ranges. See ?`trim,GenomicRanges-method` for more
  information.”
Warning message in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE):
“GRanges object contains 3 out-of-bound ranges located on sequences
  chr12, chr13, and chr16. Note that ranges located on a sequence whose
  length is unknown (NA) or on a circular sequence are not considered
  out-of-bound (use seqlengths() and isCircular() to get the lengths and
  circularity flags of the underly

In [15]:
ArchRProject.filt <- addPeakMatrix(ArchRProject.filt)

ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-198b614706362-Date-2021-12-12_Time-12-53-15.log
If there is an issue, please report to github with logFile!

2021-12-12 12:53:15 : Batch Execution w/ safelapply!, 0 mins elapsed.

.createArrowGroup : Arrow Group already exists! Dropping Group from ArrowFile! This will take ~10-30 seconds!

.dropGroupsFromArrow : Initializing Temp ArrowFile

.dropGroupsFromArrow : Adding Metadata to Temp ArrowFile

.dropGroupsFromArrow : Adding SubGroups to Temp ArrowFile

.dropGroupsFromArrow : Move Temp ArrowFile to ArrowFile

2021-12-12 12:54:14 : Adding rabbit_BGRGP1 to PeakMatrix for Chr (1 of 22)!, 0.022 mins elapsed.

2021-12-12 12:54:29 : Adding rabbit_BGRGP1 to PeakMatrix for Chr (2 of 22)!, 0.274 mins elapsed.

2021-12-12 12:54:44 : Adding rabbit_BGRGP1 to PeakMatrix for Chr (3 of 22)!, 0.521 mins elapsed.

2021-12-12 12:54:55 : Adding rabbit_BGRGP1 to PeakMatrix for Chr (4 of 22)!, 0.701 mins elapsed.

2021-12-12 12:55:02 : Adding rabbit_BGRGP1

2021-12-12 13:03:36 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (3 of 22)!, 0.383 mins elapsed.

2021-12-12 13:03:44 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (4 of 22)!, 0.519 mins elapsed.

2021-12-12 13:03:49 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (5 of 22)!, 0.612 mins elapsed.

2021-12-12 13:03:53 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (6 of 22)!, 0.666 mins elapsed.

2021-12-12 13:03:56 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (7 of 22)!, 0.722 mins elapsed.

2021-12-12 13:04:05 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (8 of 22)!, 0.874 mins elapsed.

2021-12-12 13:04:10 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (9 of 22)!, 0.962 mins elapsed.

2021-12-12 13:04:16 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (10 of 22)!, 1.064 mins elapsed.

2021-12-12 13:04:20 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (11 of 22)!, 1.117 mins elapsed.

2021-12-12 13:04:24 : Adding rabbit_BGRGP2 to PeakMatrix for Chr (12 of 22)!, 1.185 mins elapsed.

2021-12-12 13:04:

2021-12-12 13:12:32 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (10 of 22)!, 0.935 mins elapsed.

2021-12-12 13:12:35 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (11 of 22)!, 0.985 mins elapsed.

2021-12-12 13:12:39 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (12 of 22)!, 1.047 mins elapsed.

2021-12-12 13:12:46 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (13 of 22)!, 1.165 mins elapsed.

2021-12-12 13:12:54 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (14 of 22)!, 1.308 mins elapsed.

2021-12-12 13:13:01 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (15 of 22)!, 1.424 mins elapsed.

2021-12-12 13:13:07 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (16 of 22)!, 1.513 mins elapsed.

2021-12-12 13:13:12 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (17 of 22)!, 1.598 mins elapsed.

2021-12-12 13:13:17 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (18 of 22)!, 1.686 mins elapsed.

2021-12-12 13:13:22 : Adding rabbit_BGRGP8 to PeakMatrix for Chr (19 of 22)!, 1.768 mins elapsed.

2021-12-12

In [16]:
saveArchRProject(ArchRProj = ArchRProject.filt)

Saving ArchRProject...

Loading ArchRProject...

Successfully loaded ArchRProject!


                                                   / |
                                                 /    \
            .                                  /      |.
            \\\                              /        |.
              \\\                          /           `|.
                \\\                      /              |.
                  \                    /                |\
                  \\#####\           /                  ||
                ==###########>      /                   ||
                 \\##==......\    /                     ||
            ______ =       =|__ /__                     ||      \\\
        ,--' ,----`-,__ ___/'  --,-`-===================##========>
       \               '        ##_______ _____ ,--,__,=##,__   ///
        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/
        -,____,---'       \\####\\________________,--\\_##,/
         

class: ArchRProject 
outputDirectory: /rds/project/rds-SDzz0CATGms/users/bt392/04_Rabbit_ATAC_2/ArchR/Project 
samples(8): rabbit_BGRGP1 rabbit_BGRGP2 ... rabbit_BGRGP7 rabbit_BGRGP8
sampleColData names(1): ArrowFiles
cellColData names(25): Clusters sample ... ReadsInPeaks FRIP
numberOfCells(1): 33115
medianTSS(1): 3.407
medianFrags(1): 31914

In [17]:
markersPeaks <- getMarkerFeatures(
    ArchRProj = ArchRProject.filt, 
    useMatrix = "PeakMatrix", 
    groupBy = "celltype_manual",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

ArchR logging to : ArchRLogs/ArchR-getMarkerFeatures-198b6387c21e5-Date-2021-12-12_Time-13-15-24.log
If there is an issue, please report to github with logFile!

MatrixClass = Sparse.Integer.Matrix

2021-12-12 13:15:24 : Matching Known Biases, 0.009 mins elapsed.

2021-12-12 13:15:30 : Computing Pairwise Tests (1 of 13), 0.111 mins elapsed.

Pairwise Test CYT : Seqnames chr1

Pairwise Test CYT : Seqnames chr10

Pairwise Test CYT : Seqnames chr11

Pairwise Test CYT : Seqnames chr12

Pairwise Test CYT : Seqnames chr13

Pairwise Test CYT : Seqnames chr14

Pairwise Test CYT : Seqnames chr15

Pairwise Test CYT : Seqnames chr16

Pairwise Test CYT : Seqnames chr17

Pairwise Test CYT : Seqnames chr18

Pairwise Test CYT : Seqnames chr19

Pairwise Test CYT : Seqnames chr2

Pairwise Test CYT : Seqnames chr20

Pairwise Test CYT : Seqnames chr21

Pairwise Test CYT : Seqnames chr3

Pairwise Test CYT : Seqnames chr4

Pairwise Test CYT : Seqnames chr5

Pairwise Test CYT : Seqnames chr6

Pairwise Test 

Pairwise Test Megakaryocytes : Seqnames chr20

Pairwise Test Megakaryocytes : Seqnames chr21

Pairwise Test Megakaryocytes : Seqnames chr3

Pairwise Test Megakaryocytes : Seqnames chr4

Pairwise Test Megakaryocytes : Seqnames chr5

Pairwise Test Megakaryocytes : Seqnames chr6

Pairwise Test Megakaryocytes : Seqnames chr7

Pairwise Test Megakaryocytes : Seqnames chr8

Pairwise Test Megakaryocytes : Seqnames chr9

Pairwise Test Megakaryocytes : Seqnames chrX

2021-12-12 13:23:08 : Computing Pairwise Tests (9 of 13), 7.741 mins elapsed.

Pairwise Test Mesoderm : Seqnames chr1

Pairwise Test Mesoderm : Seqnames chr10

Pairwise Test Mesoderm : Seqnames chr11

Pairwise Test Mesoderm : Seqnames chr12

Pairwise Test Mesoderm : Seqnames chr13

Pairwise Test Mesoderm : Seqnames chr14

Pairwise Test Mesoderm : Seqnames chr15

Pairwise Test Mesoderm : Seqnames chr16

Pairwise Test Mesoderm : Seqnames chr17

Pairwise Test Mesoderm : Seqnames chr18

Pairwise Test Mesoderm : Seqnames chr19

Pairwise 

In [18]:
saveRDS(markersPeaks, file.path(io$outdir, 'markersPeaks.rds'))

In [20]:
# Motif enrichment
ArchRProject.filt <- addMotifAnnotations(ArchRProj = ArchRProject.filt, motifSet = "cisbp", name = "Motif", species='Homo sapiens', force=TRUE)

ArchR logging to : ArchRLogs/ArchR-addMotifAnnotations-198b6705fff93-Date-2021-12-12_Time-13-30-57.log
If there is an issue, please report to github with logFile!

peakAnnotation name already exists! Overriding.

2021-12-12 13:30:57 : Gettting Motif Set, Species : Homo sapiens, 0.003 mins elapsed.

Using version 2 motifs!

2021-12-12 13:31:01 : Finding Motif Positions with motifmatchr!, 0.065 mins elapsed.

2021-12-12 13:41:16 : Creating Motif Overlap Matrix, 10.319 mins elapsed.

2021-12-12 13:41:24 : Finished Getting Motif Info!, 10.445 mins elapsed.

ArchR logging successful to : ArchRLogs/ArchR-addMotifAnnotations-198b6705fff93-Date-2021-12-12_Time-13-30-57.log



In [21]:
saveArchRProject(ArchRProj = ArchRProject.filt)

Saving ArchRProject...

Loading ArchRProject...

Successfully loaded ArchRProject!


                                                   / |
                                                 /    \
            .                                  /      |.
            \\\                              /        |.
              \\\                          /           `|.
                \\\                      /              |.
                  \                    /                |\
                  \\#####\           /                  ||
                ==###########>      /                   ||
                 \\##==......\    /                     ||
            ______ =       =|__ /__                     ||      \\\
        ,--' ,----`-,__ ___/'  --,-`-===================##========>
       \               '        ##_______ _____ ,--,__,=##,__   ///
        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/
        -,____,---'       \\####\\________________,--\\_##,/
         

class: ArchRProject 
outputDirectory: /rds/project/rds-SDzz0CATGms/users/bt392/04_Rabbit_ATAC_2/ArchR/Project 
samples(8): rabbit_BGRGP1 rabbit_BGRGP2 ... rabbit_BGRGP7 rabbit_BGRGP8
sampleColData names(1): ArrowFiles
cellColData names(25): Clusters sample ... ReadsInPeaks FRIP
numberOfCells(1): 33115
medianTSS(1): 3.407
medianFrags(1): 31914

In [22]:
enrichMotifs <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = ArchRProject.filt,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )
saveRDS(enrichMotifs, file.path(io$outdir, 'enrichMotifs_up.rds'))

enrichMotifs <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = ArchRProject.filt,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.5"
  )
saveRDS(enrichMotifs, file.path(io$outdir, 'enrichMotifs_down.rds'))

ArchR logging to : ArchRLogs/ArchR-peakAnnoEnrichment-198b672d6e64-Date-2021-12-12_Time-13-41-38.log
If there is an issue, please report to github with logFile!

2021-12-12 13:41:54 : Computing Enrichments 1 of 13, 0.269 mins elapsed.

2021-12-12 13:41:56 : Computing Enrichments 2 of 13, 0.296 mins elapsed.

2021-12-12 13:41:57 : Computing Enrichments 3 of 13, 0.319 mins elapsed.

2021-12-12 13:41:59 : Computing Enrichments 4 of 13, 0.342 mins elapsed.

2021-12-12 13:42:00 : Computing Enrichments 5 of 13, 0.368 mins elapsed.

2021-12-12 13:42:01 : Computing Enrichments 6 of 13, 0.391 mins elapsed.

2021-12-12 13:42:03 : Computing Enrichments 7 of 13, 0.413 mins elapsed.

2021-12-12 13:42:04 : Computing Enrichments 8 of 13, 0.44 mins elapsed.

2021-12-12 13:42:06 : Computing Enrichments 9 of 13, 0.46 mins elapsed.

2021-12-12 13:42:07 : Computing Enrichments 10 of 13, 0.482 mins elapsed.

2021-12-12 13:42:08 : Computing Enrichments 11 of 13, 0.505 mins elapsed.

2021-12-12 13:42:10 : Co

In [23]:
ArchRProject.filt <- addBgdPeaks(ArchRProject.filt, method='Archr') # method ArchR bc of error ?

Identifying Background Peaks!



In [24]:
ArchRProject.filt <- addDeviationsMatrix(
  ArchRProj = ArchRProject.filt, 
  peakAnnotation = "Motif",
  force = TRUE
)

Using Previous Background Peaks!

ArchR logging to : ArchRLogs/ArchR-addDeviationsMatrix-198b61069dcea-Date-2021-12-12_Time-13-56-03.log
If there is an issue, please report to github with logFile!



NULL


2021-12-12 13:56:10 : Batch Execution w/ safelapply!, 0 mins elapsed.

.createArrowGroup : Arrow Group already exists! Dropping Group from ArrowFile! This will take ~10-30 seconds!

.dropGroupsFromArrow : Initializing Temp ArrowFile

.dropGroupsFromArrow : Adding Metadata to Temp ArrowFile

.dropGroupsFromArrow : Adding SubGroups to Temp ArrowFile

.dropGroupsFromArrow : Move Temp ArrowFile to ArrowFile

2021-12-12 13:57:03 : chromVAR deviations rabbit_BGRGP1 (1 of 8) Schep (2017), 0.88 mins elapsed.

2021-12-12 14:02:03 : rabbit_BGRGP1 (1 of 8) : Deviations for Annotation 43 of 870, 4.098 mins elapsed.

2021-12-12 14:05:50 : rabbit_BGRGP1 (1 of 8) : Deviations for Annotation 86 of 870, 7.884 mins elapsed.

2021-12-12 14:09:03 : rabbit_BGRGP1 (1 of 8) : Deviations for Annotation 129 of 870, 11.101 mins elapsed.

2021-12-12 14:13:14 : rabbit_BGRGP1 (1 of 8) : Deviations for Annotation 172 of 870, 15.292 mins elapsed.

2021-12-12 14:23:05 : rabbit_BGRGP1 (1 of 8) : Deviations for Annotat

2021-12-12 16:44:26 : chromVAR deviations rabbit_BGRGP2 (4 of 8) Schep (2017), 168.265 mins elapsed.

2021-12-12 16:47:16 : rabbit_BGRGP2 (4 of 8) : Deviations for Annotation 43 of 870, 2.325 mins elapsed.

2021-12-12 16:49:29 : rabbit_BGRGP2 (4 of 8) : Deviations for Annotation 86 of 870, 4.551 mins elapsed.

2021-12-12 16:51:22 : rabbit_BGRGP2 (4 of 8) : Deviations for Annotation 129 of 870, 6.429 mins elapsed.

2021-12-12 16:53:49 : rabbit_BGRGP2 (4 of 8) : Deviations for Annotation 172 of 870, 8.881 mins elapsed.

2021-12-12 16:56:49 : rabbit_BGRGP2 (4 of 8) : Deviations for Annotation 215 of 870, 11.881 mins elapsed.

2021-12-12 16:59:34 : rabbit_BGRGP2 (4 of 8) : Deviations for Annotation 258 of 870, 14.631 mins elapsed.

2021-12-12 17:01:58 : rabbit_BGRGP2 (4 of 8) : Deviations for Annotation 301 of 870, 17.036 mins elapsed.

2021-12-12 17:04:03 : rabbit_BGRGP2 (4 of 8) : Deviations for Annotation 344 of 870, 19.11 mins elapsed.

2021-12-12 17:06:38 : rabbit_BGRGP2 (4 of 8) : De

2021-12-12 18:56:12 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 129 of 870, 4.912 mins elapsed.

2021-12-12 18:58:05 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 172 of 870, 6.788 mins elapsed.

2021-12-12 19:00:23 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 215 of 870, 9.094 mins elapsed.

2021-12-12 19:02:30 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 258 of 870, 11.205 mins elapsed.

2021-12-12 19:04:21 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 301 of 870, 13.05 mins elapsed.

2021-12-12 19:05:56 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 344 of 870, 14.64 mins elapsed.

2021-12-12 19:07:55 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 387 of 870, 16.625 mins elapsed.

2021-12-12 19:09:22 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 430 of 870, 18.077 mins elapsed.

2021-12-12 19:10:49 : rabbit_BGRGP8 (7 of 8) : Deviations for Annotation 473 of 870, 19.524 mins elapsed.

2021-12-12 19:12:15 : rabbit_BGRGP8 (7 of 

In [26]:
saveArchRProject(ArchRProj = ArchRProject.filt)

Saving ArchRProject...

Loading ArchRProject...

Successfully loaded ArchRProject!


                                                   / |
                                                 /    \
            .                                  /      |.
            \\\                              /        |.
              \\\                          /           `|.
                \\\                      /              |.
                  \                    /                |\
                  \\#####\           /                  ||
                ==###########>      /                   ||
                 \\##==......\    /                     ||
            ______ =       =|__ /__                     ||      \\\
        ,--' ,----`-,__ ___/'  --,-`-===================##========>
       \               '        ##_______ _____ ,--,__,=##,__   ///
        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/
        -,____,---'       \\####\\________________,--\\_##,/
         

class: ArchRProject 
outputDirectory: /rds/project/rds-SDzz0CATGms/users/bt392/04_Rabbit_ATAC_2/ArchR/Project 
samples(8): rabbit_BGRGP1 rabbit_BGRGP2 ... rabbit_BGRGP7 rabbit_BGRGP8
sampleColData names(1): ArrowFiles
cellColData names(25): Clusters sample ... ReadsInPeaks FRIP
numberOfCells(1): 33115
medianTSS(1): 3.407
medianFrags(1): 31914

### plot functions

markerList <- getMarkers(markersPeaks, cutOff = "FDR <= 0.01 & Log2FC >= 1")
markerList

heatmapPeaks <- markerHeatmap(
  seMarker = markersPeaks, 
  cutOff = "FDR <= 0.1 & Log2FC >= 0.5",
  transpose = TRUE, plotLog2FC = TRUE
)

draw(heatmapPeaks, heatmap_legend_side = "bot", annotation_legend_side = "bot")

  
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

plotVarDev <- getVarDeviations(ArchRProject.filt, name = "MotifMatrix", plot = TRUE)